src/main/java/org/testng/internal/Invoker.java

```diff
@@ -1123,153 +907,142 @@ private ParameterBag createParameters(ITestNGMethod testMethod,
     final ITestClass testClass= testMethod.getTestClass();
     final List<ITestResult> result = Lists.newArrayList();
     final FailureContext failure = new FailureContext();
-    final ITestNGMethod[] beforeMethods = filterMethods(testClass, testClass.getBeforeTestMethods(), CAN_RUN_FROM_CLASS);
-    final ITestNGMethod[] afterMethods = filterMethods(testClass, testClass.getAfterTestMethods(), CAN_RUN_FROM_CLASS);
-    while(invocationCount-- > 0) {
-      if(false) {
-        // Prevent code formatting
-      }
-      //
-      // No threads, regular invocation
-      //
-      else {
-        // Used in catch statement
-        long start = System.currentTimeMillis();
-
-        Map<String, String> allParameterNames = Maps.newHashMap();
-        ParameterBag bag = createParameters(testMethod,
-            parameters, allParameterNames, suite, testContext, instance);
-
-        if (bag.hasErrors()) {
-          final ITestResult tr = bag.errorResult;
-          Throwable throwable = tr.getThrowable();
-          if (throwable instanceof TestNGException) {
-            tr.setStatus(ITestResult.FAILURE);
-            m_notifier.addFailedTest(testMethod, tr);
-          } else {
-            tr.setStatus(ITestResult.SKIP);
-            m_notifier.addSkippedTest(testMethod, tr);
-          }
-          runTestListeners(tr);
-          result.add(tr);
-          continue;
+    final ITestNGMethod[] beforeMethods = TestNgMethodUtils.filterBeforeTestMethods(testClass, CAN_RUN_FROM_CLASS);
```

Comment:
These are being passed in as parameters. I am guessing there's nothing wrong in ensuring that the variable doesn't get re-initialised by adding the `final` guard no?

Reply:
It's not important because it just a question of the code convention (which doesn't exist :p).
But as we don't use final most of the time (except class attributes), I don't see a reason to keep them.

Reply:
Fixed.